osu.Game.Rulesets.Mania/Mods/ManiaModInvert.cs

```diff
@@ -39,24 +44,34 @@ public void ApplyToBeatmap(IBeatmap beatmap)
             {
                 var newColumnObjects = new List<ManiaHitObject>();
 
-                var locations = column.OfType<Note>().Select(n => (startTime: n.StartTime, samples: n.Samples))
-                                      .Concat(column.OfType<HoldNote>().SelectMany(h => new[]
-                                      {
-                                          (startTime: h.StartTime, samples: h.GetNodeSamples(0)),
-                                          (startTime: h.EndTime, samples: h.GetNodeSamples(1))
-                                      }))
-                                      .OrderBy(h => h.startTime).ToList();
+                List<(double startTime, IList<HitSampleInfo> samples, string type)> locations;
+
+                if (FullInvert.Value)
+                    locations = column.OfType<Note>().Select(n => (startTime: n.StartTime, samples: n.Samples, type: "note"))
+                        .Concat(column.OfType<HoldNote>().SelectMany(h => new[]
+                        {
+                            (startTime: h.StartTime, samples: h.GetNodeSamples(0), type: "release"),
+                            (startTime: h.EndTime, samples: h.GetNodeSamples(1), type: "note")
+                        }))
+                        .OrderBy(h => h.startTime).ToList();
+                else
+                    locations = column.Select(n => (startTime: n.StartTime, samples: n.Samples, type: "note"))
+                        .OrderBy(h => h.startTime).ToList();
 
                 for (int i = 0; i < locations.Count - 1; i++)
                 {
-                    // Full duration of the hold note.
-                    double duration = locations[i + 1].startTime - locations[i].startTime;
+                    if (locations[i].type == "release")
+                        continue;
```

Comment:
doing this changes the behaviour of the mod completely, no? even if `FullInvert` is on?

this PR really needs a proper description because i have no idea what anything is here, and i shouldn't have to reverse-engineer the purpose of everything here from first principles